Source/WTF/wtf/posix/FileSystemPOSIX.cpp

```diff
@@ -327,17 +327,17 @@ bool makeAllDirectories(const String& path)
     if (!access(fullPath.data(), F_OK))
         return true;
 
-    char* p = fullPath.mutableData() + 1;
+    auto p = fullPath.mutableSpanIncludingNullTerminator().subspan(1);
     if (p[length - 1] == '/')
         p[length - 1] = '\0';
-    for (; *p; ++p) {
-        if (*p == '/') {
-            *p = '\0';
+    for (; p[0]; p = p.subspan(1)) {
```

Comment:
Doesn't seem like we have a strong guarantee that `p` isn't past the null terminator. Only the length check above guarantees it.

Reply:
`p[0]` would RELEASE_ASSERT.

Reply:
Good point. I was mistaken here.